docs/advanced.rst

```diff
@@ -758,6 +758,35 @@ Please refer to the supplemental example for details.
     length queries (``__len__``), iterators (``__iter__``), the slicing
     protocol and other kinds of useful operations.
 
+C++11 chrono datatypes
+======================
+When including the additional header file :file:`pybind11/chrono.h` conversions from c++11 chrono datatypes
+to corresponding python datetime objects are automatically enabled.
+The following rules describe how the conversions are applied.
+
+When passed to python objects of type ``std::chrono::system_clock::time_point`` are converted into datetime.datetime objects.
+These objects are those that specifically come from the system_clock as this is the only clock that measures wall time.
+
+When passed to python of type ``std::chrono::[other_clock]::time_point`` are converted into datetime.timedelta objects.
+These objects are those that come from all clocks that are not the system_clock (e.g. steady_clock).
+Clocks other than the system_clock are not measured from wall date/time and instead have any start time
+(often when the computer was turned on).
+Therefore as these clocks can only measure time from an arbitrary start point they are represented as timedelta from this start point.
+
+When passed to python of type ``std::chrono::duration`` are converted into datetime.timedelta objects.
+
+When python objects are passed to c++ for the case of non system clocks and durations instances of both datetime.timedelta
+and float are accepted. The float arguments are interpreted as a number of seconds since the epoch.
+
+.. note::
+
+    Other clocks may be the same as system_clock. For example on many platforms std::high_resolution_clock is the same as system_clock.
+    Because of this if you are converting a timepoint from one of these clocks they may appear to python as a datetime.datetime object.
+
+    Pythons datetime implementation is limited to microsecond precision.
+    The extra precision that c++11 clocks can have have (nanoseconds) will be lost upon conversion.
+    The rounding policy from c++ to python is via ``std::chrono::duration_cast<>`` (rounding towards 0 in microseconds).
```

Comment:
Have to say I agree with @wjakob here: the documentation reads a bit like a point-form list of things, rather than a textual description of the conversion.  Even adding just a first paragraph that gives an overview (without specific details) of what the chrono conversion support does would help a lot.

It's also missing the conversion from float values to durations/time-points, which is designed to be used with float inputs that come from python's `time.monotonic()`, `time.perf_counter()`, `time.process_time()`.